### ISSUE TITLE
puppet-lint failures should say which file it errored on

### ISSUE THREAD

**Issue by jvperrin** [chriskuehl/puppet-pre-commit-hooks#9]
I get this output when running pre-commit 1.3.0 (with hooks installed) and release v2.0.1 of this repo.

```
puppet-lint...........................................................Failed
hookid: puppet-lint

WARNING: variable not enclosed in {} on line 55
```

Ideally, the `puppet-lint` check would say which file this error occurred on. In this case, it was pretty easy to fix, since there was only one file that had a changed line 55 in this commit, but if not it would have been a bit harder to figure out.

**Comment by chriskuehl** [comment]
It looks like puppet-lint prints the filename if you pass more than one file on the command line: https://github.com/rodjek/puppet-lint/blob/536d9d53c6a4f9ae3764b3358dea8a3993c4b7b0/lib/puppet-lint/bin.rb#L52

We should be able to make it always do this by adding `--with-filename` to the args: https://github.com/rodjek/puppet-lint/blob/0799152d9ded067e1a3c5b2fa4f2748a30fbf2c5/lib/puppet-lint/optparser.rb#L36

I also wonder if we should add `--with-context` by default, that seems pretty cool:
```
$ puppet-lint --fail-on-warnings --with-filename test.pp --with-context
test.pp - WARNING: double quoted string containing no variables on line 3

  content => "foo",
             ^

test.pp - WARNING: ensure found on line but it's not the first attribute on line 4

  ensure  => present,
  ^
```

so tl;dr you could fix this in your repos by adding that to the `args` you define, but we should probably update the default args in the hooks here (and the README) too

**Comment by vchepkov** [comment]
FYI, I have solved this via parameters to the linter, here is my config

```
    - id: puppet-lint
      additional_dependencies:
        - voxpupuli-puppet-lint-plugins:~>5
      args:
        - --fail-on-warnings
        - --no-140chars-check
        - --no-documentation-check
        - --no-manifest_whitespace_opening_brace_after-check
        - --no-parameter_documentation
        - --relative
        - --with-filename
```

You are interested in the last argument
